src/main/org/codehaus/groovy/classgen/asm/sc/StaticInvocationWriter.java

```diff
@@ -602,7 +602,7 @@ private static void pushZero(final MethodVisitor mv, final ClassNode type) {
         boolean isInt = ClassHelper.int_TYPE.equals(type);
         boolean isShort = ClassHelper.short_TYPE.equals(type);
         boolean isByte = ClassHelper.byte_TYPE.equals(type);
-        if (isInt || isShort || isByte) {
+        if (isInt || isShort || isByte || ClassHelper.boolean_TYPE.equals(type)) {
```

Comment:
actuallly, the  "|| ClassHelper.boolean_TYPE.equals(type)" should be removed. isByte contains this information already.